lib/prayermate_api.rb

```diff
@@ -77,18 +76,32 @@ def post_with_auth(endpoint, data)
       process_response response
     end
 
-    def put_with_auth(endpoint)
-      response = HTTParty.put(api_path(endpoint), headers: http_headers, basic_auth: auth_hash)
+    def put_with_auth(endpoint, data = nil)
+      response = HTTParty.put(api_path(endpoint), body: data, headers: http_headers, basic_auth: auth_hash)
       process_response response
     end
 
-    def get_with_auth(endpoint)
-      response = HTTParty.get(api_path(endpoint), headers: http_headers, basic_auth: auth_hash)
+    def get_with_auth(endpoint, data = nil)
+      response = HTTParty.get(api_path(endpoint), body: data, headers: http_headers, basic_auth: auth_hash)
       process_response response
     end
 
     protected
 
+    def register(data, first_name, last_name, email, roles)
```

Comment:
I kind of get the idea of just doing this as a breaking change and being done with it... but one could also say `data = { promo_code: data } if data.is_a?(String)` and just roll it out a bit more safely?

Reply:
I think we should just roll it out as a breaking change. All the dependencies are tied to specific versions of the repo/tags as opposed to master so it shouldn't make a different to existing code (e.g. we don't need to update `prayer_processor`'s dependency